plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/SyncPartitionMetadataProcedure.java

```diff
@@ -137,8 +137,13 @@ private void doSyncPartitionMetadata(ConnectorSession session, ConnectorAccessCo
 
         try {
             FileSystem fileSystem = hdfsEnvironment.getFileSystem(hdfsContext, tableLocation);
-            List<String> partitionsInMetastore = metastore.getPartitionNames(schemaName, tableName)
+            List<String> partitionsNamesInMetastore = metastore.getPartitionNames(schemaName, tableName)
                     .orElseThrow(() -> new TableNotFoundException(schemaTableName));
+            List<String> partitionsInMetastore = metastore.getPartitionsByNames(schemaName, tableName, partitionsNamesInMetastore).values().stream()
+                    .filter(Optional::isPresent).map(Optional::get)
+                    .map(partition -> new Path(partition.getStorage().getLocation()).toUri())
+                    .map(uri -> tableLocation.toUri().relativize(uri).getPath())
```

Comment:
`sync_partition_metadata` assumes the partitions names match locations
is it still well-defined if they don't?

Reply:
I'm not sure I understand the question.
The logic here is relatively similar to what is being done in `io.trino.plugin.hive.HiveSplitManager` where the partition names are retrieved first and then the corresponding partitions are retrieved  afterwards by names.

Reply:
Yes, `sync_partition_metadata` relies on certain naming _convention_. 
Split manager shouldn't and doesn't rely on that convention.

Reply:
What changes here? Were the partition names returned as escaped or unescaped? In the new test you added, the locations in HDFS are already escaped.